internal/pubsub/samples/gcmsg/main.go

```diff
@@ -0,0 +1,224 @@
+// sends them to stdout. The name gcmsg is short for Go Cloud Messages.
+package main
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"regexp"
+
+	"github.com/google/subcommands"
+	"gocloud.dev/internal/pubsub"
+)
+
+type pubCmd struct{}
+
+func (*pubCmd) Name() string     { return "pub" }
+func (*pubCmd) Synopsis() string { return "Publish a message to a topic" }
+func (*pubCmd) Usage() string {
+	return `pub <topic>:
+  Read messages from stdin, one per line and send them to <topic>.
+`
+}
+
+func (p *pubCmd) SetFlags(f *flag.FlagSet) {
+}
+
+func (p *pubCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface{}) subcommands.ExitStatus {
+	if f.NArg() != 1 {
+		f.Usage()
+		return subcommands.ExitUsageError
+	}
+	topicURL := f.Arg(0)
+	if err := p.pub(ctx, topicURL); err != nil {
+		log.Print(err)
+		return subcommands.ExitFailure
+	}
+	return subcommands.ExitSuccess
+}
+
+func (p *pubCmd) pub(ctx context.Context, topicURL string) error {
+	t, cleanup, err := openTopic(ctx, topicURL)
+	if err != nil {
+		return fmt.Errorf("opening topic: %v", err)
+	}
+	defer cleanup()
+
+	// Get lines from stdin and send them as messages to the topic.
+	fmt.Fprintf(os.Stderr, "Enter messages, one per line to be published to \"%s\".\n", topicURL)
+	scanner := bufio.NewScanner(os.Stdin)
+	for scanner.Scan() {
+		line := scanner.Text()
+		m := &pubsub.Message{Body: []byte(line)}
+		if err := t.Send(ctx, m); err != nil {
+			return err
+		}
+	}
+	return scanner.Err()
+}
+
+// openTopic opens a topic on a supported pubsub provider, given a topic URL in a format
+// specific to gcmsg.
+func openTopic(ctx context.Context, topicURL string) (*pubsub.Topic, func(), error) {
+	u, err := url.Parse(topicURL)
+	if err != nil {
+		return nil, nil, fmt.Errorf("parsing URL: %v", err)
+	}
+	rightPart := topicURL[len(u.Scheme+"://"):]
+	if rightPart == "" {
+		return nil, nil, errors.New("empty contents in topic URL")
+	}
+	switch u.Scheme {
+	case "gcppubsub":
+		rx, err := regexp.Compile(`^projects/([^/]+)/topics/([^/]+)$`)
+		if err != nil {
+			panic(fmt.Sprintf("gcppubsub topic regex failed to copile: %v", err))
+		}
+		m := rx.FindStringSubmatch(rightPart)
+		if len(m) != 3 {
+			return nil, nil, fmt.Errorf(`gcppubsub topic URL contents "%s" failed to match "%s"`, rightPart, rx)
+		}
+		proj := m[1]
+		topicID := m[2]
+		return openGCPTopic(ctx, proj, topicID)
+	case "rabbitpubsub":
+		rx, err := regexp.Compile(`^(\w+:\w+@\w+:\d+)/topics/([^ /]+)$`)
+		if err != nil {
+			panic(fmt.Sprintf(`rabbitpubsub topic regex failed to compile: %v`, err))
+		}
+		m := rx.FindStringSubmatch(rightPart)
+		if len(m) != 3 {
+			return nil, nil, fmt.Errorf(`rabbitpubsub topic URL contents "%s" failed to match "%s"`, rightPart, rx)
+		}
+		serverURL := "amqp://" + m[1]
+		topicID := m[2]
+		return openRabbitTopic(serverURL, topicID)
+	case "":
+		return nil, nil, fmt.Errorf(`scheme missing from URL: "%s"`, topicURL)
```

Comment:
Replace
  "%s"
with
  %q

Then you can use ordinary "" for the string instead of backquotes.

Reply:
%q also escapes characters within the string, but I don't want that.